src/getters/crunchbase.py

```diff
@@ -0,0 +1,132 @@
+"""
+discovery_utils.src.getters.crunchbase.py
+
+Module for easy access to downloaded CB data on S3.
+
+"""
+import pandas as pd
+import boto3
+import os
+from dotenv import load_dotenv
+from botocore.client import BaseClient
+from src.utils import s3 as s3
+from src.utils import timestamps as ts
+from datetime import datetime, timedelta
+import logging
+import json
+
+# logging level
+logging.basicConfig(level=logging.INFO)
+
+load_dotenv()
+
+# Retrieve AWS file information from environment variables
+BUCKET_NAME_RAW = os.getenv("BUCKET_NAME_RAW")
+FILE_NAMES_RAW = json.loads(os.getenv("FILE_NAMES_RAW", '[]'))
+S3_PATH_RAW = os.getenv("S3_PATH_RAW")
+# Retrieve AWS credentials from environment variables
+AWS_ACCESS_KEY = os.getenv("AWS_ACCESS_KEY")
+AWS_SECRET_KEY = os.getenv("AWS_SECRET_KEY")
+
+
+# Left these here for when they are used using AWS CLI
+# os.environ["AWS_ACCESS_KEY"]
+# os.environ["AWS_SECRET_KEY"]
+
+
+# VERSATILE FUNCTION FOR GENERALISED USE
+
+def _s3_client(aws_access_key_id: str = AWS_ACCESS_KEY,
+               aws_secret_access_key: str = AWS_SECRET_KEY
+               ) -> BaseClient:
+    """Initialize S3 client"""
+    S3 = boto3.client(
+        "s3", aws_access_key_id, aws_secret_access_key
+        )
+    return S3
+
+def get_table(timestamp: str,
+              file_name: str,
+              s3_client: BaseClient,
+              file_names_raw: list = FILE_NAMES_RAW,
+              bucket_name_raw: str = BUCKET_NAME_RAW
+              ) -> pd.DataFrame:
+    """Specific table with named CB data from a specific version, dynamically selecting the exact file match."""
+    
+    # Validate file name is in the list of known file names without extensions
+    if file_name not in file_names_raw:
+        raise ValueError(f"File '{file_name}' not found in FILE_NAMES_RAW.")
+    
+    # Construct the directory path for the given timestamp
+    dir_path = S3_PATH_RAW + "Crunchbase_" + timestamp + "/"
+    
+    # Retrieve all file names in the directory
+    all_files = s3._get_bucket_filenames(bucket_name_raw, dir_path)
+    
+    # Filter for files that exactly match the file name, ignoring the extension
+    # This assumes file names in the bucket have a format like "filename.extension"
+    matched_files = [f for f in all_files if os.path.splitext(os.path.basename(f))[0] == file_name]
+    
+    # Check if there is exactly one matching file
+    if not matched_files:
+        raise FileNotFoundError(f"No file exactly matching '{file_name}' found in {dir_path}")
+    elif len(matched_files) > 1:
+        raise ValueError(f"Multiple files exactly matching '{file_name}' found in {dir_path}, requiring clarification.")
+    
+    # Assuming there's exactly one matching file, proceed to download and load it as a DataFrame
+    s3_key = matched_files[0]  # The full key of the matching file
```

Comment:
Suggestions:

- Split out file matching logic into it's own function (match_one_file or get_one_match), returning single matched file, raising the same errors.
- you could use Pathlib Path object to make this a bit more readable/clean

Reply:
Both done, thanks!